crates/registry/src/error.rs

```diff
@@ -6,22 +6,16 @@ pub enum RegistryError {
     MissingLatestTag(String),
     #[error("missing version `{0}` on package `${0}`")]
     MissingVersionRelease(String, String),
-    #[error("network error while downloading `${0}`")]
-    Network(reqwest::Error),
-    #[error("io error `${0}`")]
-    Io(std::io::Error),
+    #[error("network error while downloading `${source}`")]
+    Network {
+        #[from]
+        source: reqwest::Error,
+    },
+    #[error("io error `${source}`")]
+    Io {
+        #[from]
+        source: std::io::Error,
+    },
```

Comment:
I believe something like this does not require to create `{}` on the `map_error`.

```
    #[error("serialization error: `{0}")]
    Serialization(#[from] serde_yaml::Error),
```

Reply:
I am just about to say so 😆.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e61625a21a1e60c7c834f191f1ffe49a

Reply:
ohhh i wasn't sure if attributes were legal in that position. nice